src/Symfony/Component/Messenger/Transport/Receiver/ReceiverInterface.php

```diff
@@ -25,21 +25,15 @@ interface ReceiverInterface
     /**
      * Receive some messages to the given handler.
      *
-     * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope} containing the message.
-     * Note that this envelope can be `null` if the timeout to receive something has expired.
-     *
      * If the received message cannot be decoded, the message should not
      * be retried again (e.g. if there's a queue, it should be removed)
      * and a MessageDecodingFailedException should be thrown.
      *
      * @throws TransportException If there is an issue communicating with the transport
+     *
+     * @return Envelope[]
      */
-    public function receive(callable $handler): void;
-
-    /**
-     * Stop receiving some messages.
-     */
-    public function stop(): void;
+    public function get(): iterable;
```

Comment:
The iterable, like always, was kind of a pain, compared to just returning one. But, it *is* more flexible and end-users won't touch this method much anyways.